ZombieWar/src/main/java/com/csc422/zombiewar/factories/SurvivorArrayFactory.java

```diff
@@ -4,17 +4,17 @@
 import com.csc422.zombiewar.characters.Survivor;
 import com.csc422.zombiewar.characters.Child;
 import com.csc422.zombiewar.characters.Teacher;
+import com.csc422.zombiewar.weapons.Weapon;
 import java.util.ArrayList;
 import java.util.Random;
 
-
 public class SurvivorArrayFactory {
 
     Random rand = new Random();
     private int totalChild = 0;
     private int totalTeacher = 0;
     private int totalSoldier = 0;
-
+    private Weapon weapon;
     /**
```

Comment:
You need to access the weapon factory to get the weapon that is being assigned to a character
WeaponFactory factory = new WeaponFactory();

then in the loop that is creating the characters, you need to assign a random weapon from the factory. 
weapon = factory.getWeapon();